packages/langium/src/lsp/definition-provider.ts

```diff
@@ -67,21 +67,27 @@ export class DefaultDefinitionProvider implements DefinitionProvider {
 
     protected collectLocationLinks(sourceCstNode: CstNode, _params: DefinitionParams): MaybePromise<LocationLink[] | undefined> {
         const goToLink = this.findLink(sourceCstNode);
-        if (goToLink) {
-            return [LocationLink.create(
-                goToLink.targetDocument.textDocument.uri,
-                (goToLink.target.astNode.$cstNode ?? goToLink.target).range,
-                goToLink.target.range,
-                goToLink.source.range
-            )];
+        if (goToLink && goToLink.target.$segments) {
+            const name = this.nameProvider.getNameProperty(goToLink.target);
+            if (name) {
```

Comment:
Shall we use the whole composite node as target in case a name is not available?